clloader/scenarios/instance_incremental.py

```diff
@@ -40,14 +42,35 @@ def __init__(
             train=train
         )
 
+        self._random_state = np.random.RandomState(seed=random_seed)
+
         self._nb_tasks = self._setup(nb_tasks)
 
     def _setup(self, nb_tasks: int) -> int:
-        x, y, _ = self.cl_dataset.init(train=self.train)
+        x, y, t = self.cl_dataset.init(train=self.train)
 
-        # TODO: need to add seed + randomstate to ensure reproducibility
-        task_ids = np.random.randint(nb_tasks, size=len(y))
+        if t is None and nb_tasks <= 0:
+            raise ValueError(f"You need to specify a number of tasks > 0, not {nb_tasks}.")
+        elif t is None:
+            task_ids = self._random_state.randint(nb_tasks, size=len(y))
```

Comment:
je pense que la definition des task_ids devrait provenir d'une fonction "get_task_ids()". Comme ça ce sera facile a changer pour les classes filles

Reply:
Do we want to create a new class inheriting from `InstanceIncremental`? I'm sure why.

Or you simply want that the `_setup` method just call directly a new method `get_task_ids`?

Reply:
I suggested that _setup method call a new method get_task_ids instead of computing the task label inside _setup function

Reply:
It doesn't simply create task ids tho. In the case of asking less tasks than the default one, we also truncate x, and y.